middleware/event_stream.go

```diff
@@ -47,49 +21,64 @@ var producer = events.EventStreamProducer{Sender: &events.EventStreamSender{}}
 	It bails out if there isn't a `resource` field on the context which should
 	be a model.ToEvent() call in the handler.
 */
-func raiseEvent(eventType string) echo.MiddlewareFunc {
-	return func(next echo.HandlerFunc) echo.HandlerFunc {
-		return func(c echo.Context) error {
-			// first call the handler function (or the next middlware)
-			err := next(c)
-			if err != nil {
-				return err
-			}
+func RaiseEvent(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// first call the handler function (or the next middlware)
+		err := next(c)
+		if err != nil {
+			return err
+		}
 
-			// pull the "event" resource from the context, which needs to be set
-			// in the handler for this to work.
-			resource := c.Get("resource")
-			if resource == nil {
-				l.Log.Infof("failed to pull event resource from context - skipping raise event")
-				return nil
-			}
+		// specifically skip raising an event if this is set - usually when
+		// a create action happened but we do not want to re-raise the
+		// event.
+		if c.Get("skip_raise") != nil {
+			l.Log.Infof("skipping raise event per skip_raise set on context")
+			return nil
+		}
+
+		// pull the "event" resource from the context, which needs to be set
+		// in the handler for this to work.
+		resource := c.Get("resource")
+		if resource == nil {
+			l.Log.Infof("failed to pull event resource from context - skipping raise event")
+			return nil
+		}
 
-			// specifically skip raising an event if this is set - usually when
-			// a create action happened but we do not want to re-raise the
-			// event.
-			if c.Get("skip_raise") != nil {
-				l.Log.Infof("skipping raise event per skip_raise set on context")
+		eventType, ok := c.Get("event_type").(string)
+		if !ok {
+			l.Log.Warnf("Failed to cast event_type to string - exiting")
+			return nil
+		}
+
+		if c.Get("event_override") != nil {
```

Comment:
Is there specific usage of `event_override` ?  (or in future)

Reply:
Yes - it's only used for superkey right now. on the initial create we do not send the create message since the source isn't ready yet (thew orker has to handle it), but on update we override it and send the create message since the ARN got posted back from the worker.